File: 128-longest-consecutive-sequence/longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {
    if(nums.length === 0){
        return 0;
    }
    let maxLength: number=0;
    let numSet: Set<number>= new Set(nums);

    for(let i of numSet){
        if (!(numSet.has(i-1))){
            let current: number = i;
            let count: number =1;

            while(numSet.has(current+1)){
                count ++;
                current++;
                
            }
            maxLength = Math.max(maxLength, count);
        }
    }
    return maxLength;
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        if not nums:
            return 0

        maxLength = 0
        numSet = set(nums)

        for num in numSet:
            if (num - 1) not in numSet:
                current = num
                count = 1

                while current + 1 in numSet:
                    current += 1
                    count += 1

                maxLength = max(maxLength, count)

        return maxLength
